### PR TITLE
net-libs/libtins: add new package

### PR DESCRIPTION
**Pull request by OwenChia**
libtins is a high-level, multiplatform C++ network packet sniffing and crafting library.
homepage: https://libtins.github.io

### CODE REVIEW

File: net-libs/libtins/libtins-3.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cxx11 +ack-tracker +wpa2 +dot11 static-libs"
+
+REQUIRED_USE="
+	wpa2? ( dot11 )
+"
+
+DEPEND="
+	ack-tracker? ( dev-libs/boost )
```
Comment (mgorny) [inline net-libs/libtins/libtins-3.5.ebuild]: Most of C/C++ libraries you link to take `:=` slot operator, to ensure that your package gets rebuilt when they change SONAME.

File: net-libs/libtins/libtins-3.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+"
+RDEPEND="${DEPEND}
+	net-libs/libpcap
+"
+
+RESTRICT="mirror"
+
+src_configure() {
+	local mycmakeargs=(
+	    -DLIBTINS_ENABLE_CXX11="$(usex cxx11)"
```
Comment (mgorny) [inline net-libs/libtins/libtins-3.5.ebuild]: We don't use flags for C++11, enable it unconditionally.

File: net-libs/libtins/libtins-3.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		-DLIBTINS_ENABLE_ACK_TRACKER="$(usex ack-tracker)"
+		-DLIBTINS_ENABLE_WPA2="$(usex wpa2)"
+		-DLIBTINS_ENABLE_DOT11="$(usex dot11)"
+		-DLIBTINS_BUILD_SHARED="$(usex !static-libs)"
```
Comment (mgorny) [inline net-libs/libtins/libtins-3.5.ebuild]: `static-libs` requires building *both* static and shared variant. Also, unless you have a very good reason to build static libs, disable them unconditionally.

File: net-libs/libtins/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>aptx945@gmail.com</email>
+		<name>Owen Chia</name>
+	</maintainer>
+	<longdescription lang="en">
```
Comment (mgorny) [inline net-libs/libtins/metadata.xml]: Add proxy-maint team.

File: net-libs/libtins/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+		libtins is a high-level, multiplatform C++ network packet sniffing and crafting library.
+
+		Its main purpose is to provide the C++ developer an easy, efficient, platform and endianess-independent way to create tools which need to send, receive and manipulate specially crafted packets.
+	</longdescription>
+	<use>
+		<flag name="cxx11">C++11 support</flag>
```
Comment (mgorny) [inline net-libs/libtins/metadata.xml]: Sort lexically.

File: net-libs/libtins/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+		<flag name="ack-tracker">TCP ACK tracking support</flag>
+		<flag name="wpa2">WPA2 decryption support</flag>
+		<flag name="dot11">IEEE 802.11 support</flag>
```
Comment (mgorny) [inline net-libs/libtins/metadata.xml]: Wouldn't `wifi` be more user-friendly name for it? It's also defined globally.

File: net-libs/libtins/libtins-3.5.ebuild [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+"
+
+DEPEND="
+	ack-tracker? ( dev-libs/boost:= )
+	net-libs/libpcap
+	wpa2? ( dev-libs/openssl:0 )
+"
+RDEPEND="${DEPEND}"
+
+RESTRICT="mirror"
```
Comment (mgorny) [inline net-libs/libtins/libtins-3.5.ebuild]: What is the reason for mirroring restriction?

File: net-libs/libtins/libtins-3.5.ebuild (hunk 1 above)
Comment (mgorny) [inline net-libs/libtins/libtins-3.5.ebuild]: Wy did you add it to boost only?

**Comment by mgorny** [comment]
You do not seem to respect `$(get_libdir)` when installing libraries.

```
Files matching a file type that is not allowed:
   usr/lib/libtins.so.3.5
 * ERROR: net-libs/libtins-3.5::gentoo failed:
 *   multilib-strict check failed!
```

**Comment by monsieurp** [comment]
@OwenChia Hi! We're going to close this PR. Please file a new and start afresh if you still want this package to land in the tree. Thanks!

**Pull request closed without merging**